### PR TITLE
Feature - increase/decrease allowance

### PR DESCRIPTION
**Pull request by brkagithub**
Updated asset create and update features, added decrease and increase allowance features:

- Wrapper around increaseAllowance and decreaseAllowance 
- Flags to create/update to skip the increaseAllowance part
- Handle skipping in blockchain service, and decreasing allowance 
- Change how errors are handled in update/create - decrease allowance for increased amount
- Check allowance before executing increaseAllowance in create and update

Let me know if you want my demo.js file where I tested all edge cases to test this locally (it's not pushed)

### CODE REVIEW

File: services/blockchain-service/blockchain-service-base.js [hunk 1]
```diff
@@ -137,22 +137,41 @@ class BlockchainServiceBase {
...
-        stepHooks.afterHook({
-            status: OPERATIONS_STEP_STATUS.INCREASE_ALLOWANCE_COMPLETED,
-        });
+            if (BigInt(allowance) < BigInt(requestData.tokenAmount)) {
+                await this.executeContractFunction(
+                    'Token',
+                    'increaseAllowance',
+                    [
+                        serviceAgreementV1Address,
+                        BigInt(requestData.tokenAmount) - BigInt(allowance),
```
Comment (zeroxbt) [inline services/blockchain-service/blockchain-service-base.js:165]: please avoid doing same calculations in multiple places (conversion and subtraction)

**Review by zeroxbt — APPROVED** [review]

**Comment by zeroxbt** [comment]
> Since the point 'Check allowance before executing increaseAllowance transaction in update and create' has been implemented, it's kind of redundant to even be able to pass in true as skipIncreaseAllowance value.
> 
> Even if you do not pass in skip=true, the logic in the code will know in case you have enough allowance to do the operation without increasing it - and will not increase it (it will skip the increasing of the allowance).
> 
> My question is - should we keep that added parameter to create, update? It's been specified in the Jira story that it should have a parameter that has false for defaultValue and it has been done this way. The only drawback to not having the parameter I see is - it's clearer for users that they are skipping the increase allowance part.
> 
> Let me know if I'm wrong and what should be done - function to take a look at is createAsset in blockchain-service-base

I agree, the parameter is not necessary and it would make sense to remove it.

**Review by zeroxbt — APPROVED** [review]

**Review by NZT48 — APPROVED** [review]
